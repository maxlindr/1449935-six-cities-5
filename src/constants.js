export const OfferCardType = {
  FAVORITE: `favorite`,
  NEAR_PLACE: `near_place`,
  CITIES: `cities`
};

export const City = {
  PARIS: `Paris`,
  COLOGNE: `Cologne`,
  BRUSSELS: `Brussels`,
  AMSTERDAM: `Amsterdam`,
  HAMBURG: `Hamburg`,
  DUSSELDORF: `Dusseldorf`
};

export const AuthorizationStatus = {
  NOT_AUTHORIZED: `NOT_AUTHORIZED`,
  AUTHORIZED: `AUTHORIZED`,
  PENDING: `PENDING`
};

export const AppRoute = {
  LOGIN: `/login`,
  ROOT: `/`,
  FAVORITES: `/favorites`,
  ERROR: `/error`,
  OFFER: `/offer/:id`,
};

export const APIRoute = {
  OFFERS: `/hotels`,
  LOGIN: `/login`,
  FAVORITE: `/favorite`
};

export const HttpCode = {
  UNAUTHORIZED: 401,
  NOT_FOUND: 404,
};

export const ErrorMessage = {
  NOT_FOUND: `Resource not found`,
  GENERAL: `Oops! Something went wrong`,
  UNAUTHORIZED: `Unauthorized`,
};
