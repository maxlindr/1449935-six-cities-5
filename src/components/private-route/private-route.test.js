import React from 'react';
import {BrowserRouter, Route} from 'react-router-dom';
import renderer from 'react-test-renderer';
import {PrivateRoute} from './private-route';
import {AppRoute, AuthorizationStatus} from '../../constants';

const MockComponent = () => <div>Mock component</div>;
const RedirectComponent = () => <div>Redirect component</div>;

const renderComponent = (authorizationStatus) => (
  renderer.create((
    <BrowserRouter>
      <PrivateRoute
        path={AppRoute.ROOT}
        authorizationStatus={authorizationStatus}
        render={MockComponent}
      />

      <Route path={AppRoute.LOGIN}>
        <RedirectComponent />
      </Route>
    </BrowserRouter>
  ))
);

describe(`PrivateRoute должен корректно отображаться`, () => {
  it(`Должен корректно отображаться при authorizationStatus=AuthorizationStatus.AUTHORIZED`, () => {
    const tree = renderComponent(AuthorizationStatus.AUTHORIZED).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`Должен редиректить на маршрут /login при authorizationStatus=AuthorizationStatus.NOT_AUTHORIZED`, () => {
    const tree = renderComponent(AuthorizationStatus.NOT_AUTHORIZED).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
