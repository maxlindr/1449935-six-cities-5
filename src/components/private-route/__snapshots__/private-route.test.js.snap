// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute должен корректно отображаться Должен корректно отображаться при authorizationStatus=AuthorizationStatus.AUTHORIZED 1`] = `
<div>
  Mock component
</div>
`;

exports[`PrivateRoute должен корректно отображаться Должен редиректить на маршрут /login при authorizationStatus=AuthorizationStatus.NOT_AUTHORIZED 1`] = `
<div>
  Redirect component
</div>
`;
